packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -306,11 +306,24 @@ class RouteSettings {
   /// The name of the route (e.g., "/settings").
   ///
   /// If null, the route is anonymous.
+  ///
+  /// The is also used to create a default semantic value for the route
```

Comment:
Does this work well since route names are usually hierarchically, e.g. "/home/login"? What would TalkBack or VoiceOver announce here?

Reply:
on Android I observed that TalkBalk splits these up - so "home/login" gets read as "home, login"  which seems like a reasonable default to me.

Reply:
I believe native Android would just announce "Login" in this case and not read the entire navigation stack? If we can't do better by default, we should at least give a pointer that tells you how to customize it if you want.

/cc @Hixie Do you have an opinion about reusing the (potentially structured) name as a default announcement?

Reply:
Some other options I explored - do string manipulation some home/login becomes [home, logic] and we only announce the last one.  That requires users to be using routes as we expect, and I don't really like magic string rules.

The other option is to add a new parameter to the route settings, but I'm not sure the best way to plumb that through, since usually we build route settings from a Map<String, WidgetBuilder>

Reply:
We could add a new param to the navigator called an "accessibilityBuilder" or something, which takes the routeSettings name and spits out the name used in the semantics node. This could default to splitting on"/" and returning the last segment

Reply:
This is an internal opaque string. It doesn't make sense to use as a user-visible string.